pkg/client/cache/delta_fifo.go

```diff
@@ -199,6 +258,7 @@ func (f *DeltaFIFO) queueActionLocked(actionType DeltaType, obj interface{}) err
 		return KeyError{obj, err}
 	}
 	newDeltas := append(f.items[id], Delta{actionType, obj})
```

Comment:
@deads2k That's not how append works-- it doesn't change the slice. It might change the memory the slice points to, but that's only important if someone else has a differently-sliced pointer to the same memory. Merely having a slice with enough capacity in the map is not enough to trigger a bug, because append doesn't change the slice that's stored in the map.

Compare: https://play.golang.org/p/mWvfcMX3mM
With: https://play.golang.org/p/aI6OSdMtPd

I don't have the bug the latter demonstrates for reasons beyond the scope of this one line-- we own the delta slice if it's in the map, we don't own it if it isn't. If I understand you correctly, though, your model of append would have printed out "[hi mom] [hi mom]", not "[hi] [hi mom]" for the first line, which is not the case. Did I misunderstand you?